bundles/org.openhab.core.thing.xml/src/main/java/org/eclipse/smarthome/core/thing/xml/internal/ThingTypeXmlProvider.java

```diff
@@ -131,34 +131,46 @@ private void addConfigDescription(ConfigDescription configDescription) {
     public synchronized void addingFinished() {
         Map<String, ChannelType> channelTypes = new HashMap<>(10);
         // create channel types
-        for (ChannelTypeXmlResult type : this.channelTypeRefs) {
+        for (ChannelTypeXmlResult type : channelTypeRefs) {
             ChannelType channelType = type.toChannelType();
-            channelTypes.put(channelType.getUID().getAsString(), channelType);
-            this.channelTypeProvider.add(this.bundle, channelType);
+            try {
+                channelTypes.put(channelType.getUID().getAsString(), channelType);
+                channelTypeProvider.add(bundle, channelType);
+            } catch (Exception e) {
+                logger.error("Could not register ChannelType: {}", channelType.getUID(), e);
```

Comment:
Better log `type.getUID()`?

```suggestion
                logger.error("Could not register ChannelType: {}", type.getUID(), e);
```

Reply:
Yes I'd prefer that too but that method doesn't exist on that object. :-)

Reply:
`ChannelTypeXmlResult` does not have such a method but `ChannelGroupTypeXmlResult` does? As well as all other `*TypeXmlResult` classes? Does it make sense to add it too? Or is that not possible?

Reply:
And I expect the exception will be thrown in the `type.toChannelType()` method which needs to be covered.

Reply:
If [doesn't have a UID field](https://github.com/openhab/openhab-core/blob/64ae35a0a354a282b6f50f99da44c2d8bfd38de2/bundles/org.openhab.core.thing.xml/src/main/java/org/eclipse/smarthome/core/thing/xml/internal/ChannelTypeXmlResult.java#L30-L32) like the other classes. The only exception that can occur with `type.toChannelType()` is a NPE when `type` is null because the `toChannelType` method is a [simple getter](https://github.com/openhab/openhab-core/blob/64ae35a0a354a282b6f50f99da44c2d8bfd38de2/bundles/org.openhab.core.thing.xml/src/main/java/org/eclipse/smarthome/core/thing/xml/internal/ChannelTypeXmlResult.java#L44-L46).

Reply:
I mean, you can return the `channelType.getUID()`... But I think that will be a little bit over the top. So just stop it here.